src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -53,56 +53,58 @@ namespace
 
     // pinvoke_override:
     // Check if given function belongs to one of statically linked libraries and return a pointer if found.
-    const void* STDMETHODCALLTYPE pinvoke_override(const char* libraryName, const char* entrypointName)
+    const void* STDMETHODCALLTYPE pinvoke_override(const char* library_name, const char* entry_point_name)
     {
+        // This function is only called with the library name specified for a p/invoke, not any variations.
+        // It must handle exact matches to the names specified. See Interop.Libraries.cs for each platform.
 #if defined(_WIN32)
-        const char* hostPolicyLib = "hostpolicy.dll";
+        const char* hostpolicy_lib_name = "hostpolicy";
 
-        if (strcmp(libraryName, "System.IO.Compression.Native") == 0)
+        if (strcmp(library_name, "System.IO.Compression.Native") == 0)
```

Comment:
Should we have a macro to prepends `lib` to a string constant on Unix and use it for both System.IO.Compression.Native and hostpolicy?


```
    if (strcmp(library_name, LIBRARY_NAME("System.IO.Compression.Native")) == 0)
    {
        return CompressionResolveDllImport(entry_point_name);
    }

#if !defined(_WIN32)
    if (strcmp(library_name, LIBRARY_NAME("System.Net.Security.Native")) == 0)  
    {
....

#endif

    if (strcmp(library_name, LIBRARY_NAME("hostpolicy")) == 0)
    {
        ...
    }
```

Reply:
We have some related macros - the existing ones include the file extension and use wide/ordinary strings based on the platform. I could look at reworking and adding some.

https://github.com/dotnet/runtime/blob/85ddef115d1c279d9e1cb315b93fa218f9dfb004/src/native/corehost/hostmisc/pal.h#L76-L77